pkg/goDB/encoder/lz4/lz4.go

```diff
@@ -2,20 +2,71 @@
 package lz4
 
 /*
-#cgo CFLAGS: -O3 -g
-#define LZ4_STATIC_LINKING_ONLY
-
+#cgo linux LDFLAGS: -llz4
+#cgo darwin,amd64 LDFLAGS: -llz4
+#cgo darwin,arm64 LDFLAGS: -llz4
 #include <stdlib.h>
 #include <stdio.h>
-#include "lz4hc.h"
-#include "lz4.h"
-
-int cCompress(int len, char *input, char *output, int level) {
-  return LZ4_compressHC2(input, output, len, level);
+#include "lz4frame.h"
+
+int cCompress(char *src, int srcSize, char *dst, int level) {
+	// initialize the frame compression preferences
+	// taken from https://github.com/lz4/lz4/blob/dev/examples/frameCompress.c
+	const LZ4F_preferences_t prefs = {
+    	{
+			LZ4F_default,
+			LZ4F_blockLinked,
+			LZ4F_noContentChecksum,
+			LZ4F_frame,
+			0,
+			0,
+			LZ4F_noBlockChecksum,
+		},
+		level,
+		0,
+		0,
+		{ 0, 0, 0 },
+	};
+
+	return LZ4F_compressFrame(dst, LZ4F_compressFrameBound(srcSize, &prefs), src, srcSize, &prefs);
```

Comment:
Probably the cleanest solution, yeah (although it's annoying because you'll have to hand over the same set of prefs from Go to C twice, which is probably not pleasant). Your call, I's also be fine just working around it by:
* Setting an absolute minimum on the allocation in Go land (after all, anything below a page size of 4096 bytes could just be, well, 4096 bytes, no real harm done).
* Increase the absolute margin of error to, i.d.k., 64 bytes instead of 16 bytes.
Probably the worst case assumption just changed over time.